src/bot/scripts/card.js

```diff
@@ -728,946 +66,1233 @@ const card = new Command({
+              }
 
-          embed.description += "\n";
-        }
+              await downloadImage(url, pathResolve);
 
-      } else {
+            }
+          })
 
-        embed.description += "**Power Information? - No**\n\n";
-      }
+          .catch(console.error);
 
-      if (card.Type === "Character" && card.CanAttack && card.CharacterType !== "Totem") {
-        // card that can attack
+        const cardId = commandValues.matchedCards._id;
 
-        embed.description += "**Can Attack? - Yes**\n";
 
-        embed.description += `Attack Range: ${Math.round(card.AttackRange * 100) / 100}\n`;
-        embed.description += `Knockback: ${Math.round(parseInt(card.KnockbackImpulse) * 100) / 100} at ${Math.round(card.KnockbackAngleDeg * 100) / 100}°\n\n`;
+        rp(`https://sppd.feinwaru.com/api/v1/cards/${cardId}`)
+          .then(async response1 => {
+            const data1 = JSON.parse(response1);
 
-      } else {
-        // spell, totem or card that cant attack
+            let cardData = data1.data;
 
-        embed.description += "**Can Attack? - No**\n\n";
-      }
+            let level = commandValues.value;
 
-      if (card.AOEAttackType !== "No" && card.Type !== "Spell") {
-        // aoe attacks
+            // card select
+            let card = data1.data;
 
-        embed.description += "**AOE Attacks? - Yes**\n";
+            // if name is %r, select a random card
 
-        embed.description += `AOE Damage Percentage: ${Math.round(card.AOEDamagePercentage * 100) / 100}\n`;
-        embed.description += `AOE Knockback Percentage: ${Math.round(card.AOEKnockbackPercentage * 100) / 100}\n`;
-        embed.description += `AOE Radius: ${Math.round(card.AOERadius * 100) / 100}\n\n`;
+            // -card <name> (art)
+            // where name is compulsory
+            // and art is optional
 
-      } else {
-        // no aoe
+            // find the card by % match
 
-        embed.description += "**AOE Attacks? - No**\n\n";
-      }
+            if (commandValues.modifier === "art") {
+              return message.channel
+                .send("", {
+                  file: path.join(
+                    __dirname,
+                    "..",
+                    "assets",
+                    "cards",
+                    "art",
+                    `${card.image}.jpg`
+                  )
+                })
+                .catch(err => {
+                  message.channel.send(`error sending card art: ${err}`);
+                });
+            }
 
-      embed.description += `Full Stats: https://sppd.feinwaru.com/${card.Image}`;
+            // friendly-fight/challenge stats
+            // ignore this if the command is -card ff,
+            // and assume 'ff' is a card name instead
+            // this needs to be called after card search
+            // as we need the rarity of the card
+
+            function _calculateCardAugmentData(original, utype, uvalue) {
```

Comment:
move to global scope, make arrow function

Reply:
done